balancer/weightedroundrobin/weightedroundrobin.go

```diff
@@ -19,11 +19,38 @@
 // Package weightedroundrobin defines a weighted roundrobin balancer.
 package weightedroundrobin
 
+import (
+	"google.golang.org/grpc/resolver"
+)
+
 // Name is the name of weighted_round_robin balancer.
 const Name = "weighted_round_robin"
 
-// AddrInfo will be stored inside Address metadata in order to use weighted roundrobin
-// balancer.
+// attributeKey is the type used as the key to store AddrInfo in the Attributes
+// field of resolver.Address.
+type attributeKey struct{}
+
+// AddrInfo will be stored inside Address metadata in order to use weighted
+// roundrobin balancer.
 type AddrInfo struct {
 	Weight uint32
 }
+
+// SetAddrInfo returns a copy of addr in which the Attributes field is updated
+// with addrInfo.
+//
+// This is an EXPERIMENTAL API.
+func SetAddrInfo(addrInfo AddrInfo, addr resolver.Address) resolver.Address {
+	addr.Attributes = addr.Attributes.WithValues(attributeKey{}, addrInfo)
+	return addr
+}
+
+// GetAddrInfo returns the AddrInfo stored in the Attributes fields of addr.
+// Returns nil if no AddrInfo is present.
```

Comment:
This is not returning nil.

Should we just return the zero value? Or also return an OK?

Reply:
It would be better if we can change `Attributes.Value` to return `(interface{}, bool)`. So, all users of that package will benefit.

Reply:
I don't understand - can you explain how this will help?

Reply:
Given that the key for `Attributes` is local to the package, if `Attributes.Value` returned an extra return parameter indicating whether or not it found the key, the callers could be less paranoid when type asserting the returned `interface{}` and simply return a default value if the second return parameter is `false`.

Reply:
The type assertion is safe, however, if you use the `, ok` form of it.  And you'll get the zero value if the `interface{}` is `nil`.  And if you do need it, you can rewrite:

```go
v, ok := a.Value(...)
if ok {
```

as

```go
v := a.Value(...)
if v != nil {
```